### ISSUE TITLE
crop doesn't work inside bootstrap modal

### ISSUE THREAD

**Issue by ashiefk** [alexk111/ngImgCrop#95]
crop doesn't work inside bootstrap modal, 
ng-show="enableCrop" this enableCrop not setting inside popup

var modalInstance = $modal.open({
		      animation: true,
		      templateUrl: 'elements/imageCrop.html',
		      controller: 'instanceController',
		      windowClass:'loginModal',
		      size: 'lg'
		    });

Please help...i really need this to be sorted out

**Comment by Sancer** [comment]
Declare ng-init="init()" on modal template

$scope.init = function() {
				var file = angular.element(document.getElementById("fileInput"));
				file.on('change', handleFileSelect)
			}

**Comment by ashiefk** [comment]
Thank you for your reply,
The issue here is ng-show="enableCrop" the browse element doesn't show up because of enableCrop is false. 
the scope of popup doesn't set inside ngImgCrop module. Please help.

**Comment by albertonaperijr** [comment]
Can you show the codes of your modal controller?

**Comment by ducnvhn** [comment]
I have the same issue on angular - bootstrap ui modal. 
here is my modal controller.
.controller('ImageModalCtrl',['$scope','$modalInstance',function($scope,$modalInstance){
            $scope.init=function(){
            angular.element(document.querySelector('#fileInput')).on('change',handleFileSelect);
            }
            $scope.myImage='';
            $scope.myCroppedImage='';

            var handleFileSelect=function(evt) {
                var file=evt.currentTarget.files[0];
                var reader = new FileReader();
                reader.onload = function (evt) {
                    console.log('load image');
                    $scope.$apply(function($scope){
                        $scope.myImage=evt.target.result;
                    });
                };
                reader.readAsDataURL(file);
            };

        }]);
HTML markup is the one I copy from the example.

**Comment by albertonaperijr** [comment]
@ducnvhn What angularjs version are you using?

**Comment by ducnvhn** [comment]
Angularjs  version 1.3.9

**Comment by albertonaperijr** [comment]
@ducnvhn As I experienced latest versions of angular doesn't work with the demo code so I've several changes as I mentioned in the other issue. Link: https://github.com/alexk111/ngImgCrop/issues/93

By the way, you can try declaring your image source and cropped image result inside an object. It's safer to declare it that way. [JSFiddle link](https://jsfiddle.net/albertonaperijr/6nwy438j/1/)


**E.g.**

**HTML:**
```javascript
<img-crop image="cropper.myImage" result-image="cropper.myCroppedImage"></img-crop>
```
**JS:**
```javascript
$scope.cropper = {
    myImage: '',
    myCroppedImage: ''
};

$scope.uploadFile = function(file) {
    if (file) {
        // ng-img-crop
        var imageReader = new FileReader();
        imageReader.onload = function(image) {
            $scope.$apply(function($scope) {
                $scope.cropper.myImage = image.target.result;
            });
        };
        imageReader.readAsDataURL(file);
    }
};
```

Hope this will help you.

**Comment by silentFred** [comment]
thanks @albertonaperijr

**Comment by albertonaperijr** [comment]
@silentFred You're welcome!

**Comment by comapedrosa** [comment]
Thanks @albertonaperijr!

**Comment by albertonaperijr** [comment]
@comapedrosa You're welcome!

**Comment by babakva** [comment]
@albertonaperijr Not sure how you got it to work, but following your solution didn't work for me. This is inside my modal before selecting an image
```
<div class="modal-body">
	<div class="cropArea">
		<img-crop image="myImage" result-image="myCroppedImage"></img-crop>
		<input type="text" class="form-control" ng-value="myImage">
		<input type="text" class="form-control" ng-value="myCroppedImage">
	</div>
</div>
```
and after selecting an image
```
<div class="modal-body">
	<div class="cropArea">
		<img-crop image="myImage" result-image="myCroppedImage"></img-crop>
		<input type="text" class="form-control" ng-value="myImage" value="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAMwAAADMCAMAAA.....">
		<input type="text" class="form-control" ng-value="myCroppedImage" value="">
	</div>
</div>
```
Note that the ng-value is set in the input but not for the other images. Any idea why?

**Comment by sureshkoduri** [comment]
Thanks @albertonaperijr

**Comment by RajanJS** [comment]
Saved my day !!! Thanks!!!

**Comment by RajanJS** [comment]
@albertonaperijr You saved my day !!!

**Comment by IssamEssarghini033** [comment]
Thank u brother

**Comment by ilaipi** [comment]
@albertonaperijr You saved my day !!!

**Comment by albertonaperijr** [comment]
@IssamEssarghini033, @ilaipi, & @sureshkoduri  You're welcome!
Guys, you can use the extended version since it is in active development: https://github.com/CrackerakiUA/ngImgCropFullExtended/
It also has rectangle option.

**Comment by sureshkoduri** [comment]
@albertonaperijr , Hi bro, I'm using it bro. Its awesome..

**Comment by anshulguleria** [comment]
None of the above solutions worked for me. But what worked was using ng-init in directive to set cropImage to empty string.
```html
<img-crop image="modal.imageObj.sourceImg"
                      result-image="modal.imageObj.cropImg"
                      area-type="square"
                      ng-init="modal.imageObj.cropImg=''"
                      result-image-size="500"></img-crop>
```
But don't know what was the problem. As it was working fine when I was using this inside normal page. But when I started using it inside angular UI bootstrap modal crop image stopped updating.

**Comment by tomasg88** [comment]
Hi,
Althugh this is an OOOOOOLLD issue, im gonna add my 5 cents as I have just spent more than 3 hours figuring out why this wasn't working for me. I thought that because my new app version used bootstrap modals, that was the reason for the plugin for not to work. Specifically, image was loading great after selecting it, but my issue was "result-image" prop not getting updated.
Ng-init didn't do solve it and changing how to handle 'onchange' event didn't either.
I was able to solve this issue by putting the "result-image" variable under a namespace, meaning, not creating it directly under $scope. Like this:

`angular.module('mod').controller('ctrl', function() {
 init()
 function init() {
 $scope.namespace = {};
 $scope.namespace.resultImage = '';
 $scope.imageSrc = '';
}
})`

Then on the view, I just did:
`
        <img-crop
          image="imageSrc"
          result-image="namespace.imageResult"
          area-type="circle">
        </img-crop>
`

Hope this helps!
